napalmpapalam/keep-a-changelog-rs
# Write compliant markdown as validated by Markdownlint (issue #3)

###  Objective 

Ensure that keep-a-changelog writes compliant markdown as validated by Markdownlint. 
Addressing issue #3 from @jerusdp. 
Add a flag permitting a compact output and disabling the header and list lints checking for blank lines.

### Testing 

Implemented data and tests based on following  four scenarios:

- Writing a default changelog as defined by keep-a-changelog defaults
- A default changelog with a single (unreleased) release
- Unreleased with a single Added entry
- Early changelog (based on the current crate ChangeLog) 
- Markdown compliant and compact form examples of each

Each of these cases are tested in two ways:

- create the markdown file using keep-a-changelog. It should be the same as the reference data.
- open the reference data and saved it to another file (read and write back) both should be the same.

### Changes

- implement test data (validated by Markdownlint) and tests for scenarios
- ensure blank lines are printed after each header, final paragraph and final bulleted list item
- remove any residual blank line at the end of the file
- ensure all test cases are passing
- compact flag produces compact form and disables lints checking headers and lists

@napalmpapalam
